tools/standalone-binary-distro/make_distro_builds.sh

```diff
@@ -21,7 +21,9 @@ OUT_FILENAME="Datadog-$SDK_VERSION.zip"
 # create temporary xcconfig for carthage build
 set -euo pipefail
 xcconfig=$(mktemp /tmp/static.xcconfig.XXXXXX)
-trap 'rm -f "$xcconfig" ;' INT TERM HUP EXIT
+# carthage may try to build schemes in dependency-manager-tests folder, so we hide it temporarily
+mv dependency-manager-tests/ .dependency-manager-tests/
+trap 'rm -f "$xcconfig" ; mv .dependency-manager-tests/ dependency-manager-tests/ ;' INT TERM HUP EXIT
```

Comment:
This is getting quite messy as we're adding more patching. It will be hard to ensure clean state before / after running this script if we don't control it clearly. How about ensuring clean state before / after running this script by introducing simple functions, e.g.:
```shell
$xcconfig = "binary_distribution.xcconfig"

# Prepares the repository folder for building xcframeworks.
function prepare() {
   echo "BUILD_LIBRARY_FOR_DISTRIBUTION = YES" >> $xcconfig
   mv dependency-manager-tests/ .dependency-manager-tests/ 
   mv Carthage .Carthage
}

# Resets repository changes made in 'prepare()'
function cleanup() {
   rm $xcconfig
   mv .dependency-manager-tests/ dependency-manager-tests/
   mv .Carthage Carthage
}

trap cleanup INT TERM HUP EXIT
prepare

# ... produce xcframeworks
```

Reply:
i made changes along those lines 👍